server/plugin/api.go

```diff
@@ -0,0 +1,1696 @@
+		return
+	}
+
+	var subscriptionList []*serializers.SubscriptionDetails
+	var subscriptionErr error
+	createdBy := r.URL.Query().Get(constants.QueryParamCreatedBy)
+	switch createdBy {
+	case constants.FilterCreatedByMe, "":
+		subscriptionList, subscriptionErr = p.Store.GetAllSubscriptions(mattermostUserID)
+	case constants.FilterCreatedByAnyone:
+		subscriptionList, subscriptionErr = p.Store.GetAllSubscriptions("")
+	}
+	if subscriptionErr != nil {
+		p.API.LogError(constants.FetchSubscriptionListError, "Error", subscriptionErr.Error())
+		p.handleError(w, r, &serializers.Error{Code: http.StatusInternalServerError, Message: subscriptionErr.Error()})
+		return
+	}
+
+	offset, limit := p.GetOffsetAndLimitFromQueryParams(r)
+	channelID := r.URL.Query().Get(constants.QueryParamChannelID)
+	serviceType := r.URL.Query().Get(constants.QueryParamServiceType)
+	eventType := r.URL.Query().Get(constants.QueryParamEventType)
+	project := r.URL.Query().Get(constants.QueryParamProject)
+	if project != "" {
+		subscriptionByProject := []*serializers.SubscriptionDetails{}
+		for _, subscription := range subscriptionList {
+			if subscription.ProjectName == project {
+				if channelID == "" || subscription.ChannelID == channelID {
+					switch serviceType {
+					case "", constants.FilterAll:
+						subscriptionByProject = append(subscriptionByProject, subscription)
+					case constants.FilterBoards:
+						switch eventType {
+						case "", constants.FilterAll:
+							if constants.ValidSubscriptionEventsForBoards[subscription.EventType] {
+								subscriptionByProject = append(subscriptionByProject, subscription)
+							}
+						default:
+							if subscription.EventType == eventType {
+								subscriptionByProject = append(subscriptionByProject, subscription)
+							}
+						}
+					case constants.FilterRepos:
+						switch eventType {
+						case "", constants.FilterAll:
+							if constants.ValidSubscriptionEventsForRepos[subscription.EventType] {
+								subscriptionByProject = append(subscriptionByProject, subscription)
+							}
+						default:
+							if subscription.EventType == eventType {
+								subscriptionByProject = append(subscriptionByProject, subscription)
+							}
+						}
+					case constants.FilterPipelines:
+						switch eventType {
+						case "", constants.FilterAll:
+							if constants.ValidSubscriptionEventsForPipelines[subscription.EventType] {
+								subscriptionByProject = append(subscriptionByProject, subscription)
+							}
+						default:
+							if subscription.EventType == eventType {
+								subscriptionByProject = append(subscriptionByProject, subscription)
+							}
+						}
+					}
+				}
+			}
+		}
+
+		sort.Slice(subscriptionByProject, func(i, j int) bool {
+			return subscriptionByProject[i].ChannelName+
+				subscriptionByProject[i].EventType+
+				subscriptionByProject[i].TargetBranch+
+				subscriptionByProject[i].PullRequestCreatedByName+
+				subscriptionByProject[i].PullRequestReviewersContainsName+
```

Comment:
Why sort them this way specifically? Maybe have a `UpdatedAt`, and sort by subscription title by default?

Reply:
Fixed, now we are sorting them based on the field `createdAt`